blockstore.go

```diff
@@ -442,8 +439,7 @@ Retry:
 		for _, block := range blocks {
 			err := txn.Put(b.db, block.Cid().Hash(), block.RawData(), lmdb.NoOverwrite)
 			if err != nil && !lmdb.IsErrno(err, lmdb.KeyExist) {
-				txn.Abort() // short-circuit
```

Comment:
is this automatically aborted?

Reply:
Yep, when returning `err != nil` the tx is aborted manually.